### ISSUE TITLE
Is it possible to define examples for individual `Model` or `Serializer` fields?

### ISSUE THREAD

**Issue by mlegner** [tfranzel/drf-spectacular#767]
I would like to define examples for individual `Model` or `Serializer` fields, similar to generating `descriptions` based on the `help_text` parameters.

Unfortunately, so far I've only found ways to defining complete examples for views and serializers based on [`OpenApiExample`](https://drf-spectacular.readthedocs.io/en/latest/drf_spectacular.html?highlight=OpenApiExample#drf_spectacular.utils.OpenApiExample) with [`extend_schema`](https://drf-spectacular.readthedocs.io/en/latest/readme.html?highlight=OpenApiExample#customization-by-using-extend-schema) and [`extend_schema_serializer`](https://drf-spectacular.readthedocs.io/en/latest/customization.html#step-4-extend-schema-serializer), respectively.

Is it possible to define examples for individual `Model` or `Serializer` fields with DRF-Spectacular and, if yes, how?

If this is not possible yet, would this be a feature that could be added? I would be happy to work on this if maintainers agree and could point me in the right direction.

**Comment by tfranzel** [comment]
Hi, I think I get what you are asking for. The thing is we settled on serializer/parameter examples as atomic units. They already have a complicated mix and match logic behind them. 

I think our interface simply would be more confusing by allowing it, even though I get the convenience of it. It is a tradeoff at the end of the day. The question that one would need to ask is what happens if a field has examples while the serializer either has attached examples or there are examples otherwise injected via `@extend_schema`. Should this override, merge, be ignored or handled otherwise then? No easy answer there imho. 

What do you think? Do you have a idea how to design it without being confusing? The implementation itself is not really the problem.

**Comment by mlegner** [comment]
Thanks for your insights, @tfranzel!

As far as I understand the [OpenAPI specification](https://swagger.io/docs/specification/adding-examples/), examples can be independently added to various entities like requests/responses, objects, and individual properties; these examples can be present at the same time. So from the point of view of drf-spectacular, the examples of individual fields could just be added to the corresponding properties in the model's (or serializer's) schema while a "global" example for the model or serializer would be an example for the object itself. We don't really have to decide how to merge/overwrite examples as this would be the task of additional systems like Swagger UI.

If we had this 1:1 correspondence, this would not be too confusing in my opinion. Users could decide for themselves at which level they want to add the examples (which is already possible now with examples for serializes and views).

**Comment by mlegner** [comment]
I just did a quick test on https://editor.swagger.io/ with the petstore example and observed that the behavior is indeed what I expected and described above: Request/response examples (completely) hide object examples which in turn (completely) hide property examples.

**Comment by tfranzel** [comment]
Excellent! Thank you for preparing a test. This that cleared up, I think we can do a PR. Would you care to give this a shot or should I add it to my todo list?

**Comment by tfranzel** [comment]
awesome! Don't be shy to ask for early feedback on the PR even if is not done yet.

**Comment by mlegner** [comment]
@tfranzel I just thought a bit more about how to implement that and here's my current idea:

It would be great if we could add the examples directly to the individual `Field`s. Unfortunately, there is no existing attribute we can (ab)use for that (like the `help_text` for the `description`). My solution here would be a function `with_example` that defines an additional `example` attribute on the field:

```python
def with_example(field: Field, example: str) -> Field:
    setattr(field, "example", example)
    return field
```

Then we can just extract this within `DjangoFilterExtension.resolve_filter_field`, similar to how we set the `description`.

Does this sound reasonable to you?

**Comment by tfranzel** [comment]
> Unfortunately, there is no existing attribute we can (ab)use for that ...

yeah, and DRF is bitchy about adding params it does not know about. we tried that before.

I don't really like adding custom methods, and thereby extending DRF's API, as we got away without with everything else until this point. I would really like to stay inside the existing idioms we created for spectacular so users don't need to learn another thing. That is not a hard no, but we should exhaust the other options first.

**Comment by diego351** [comment]
Allow me to show you this one:

<img width="585" alt="image" src="https://user-images.githubusercontent.com/1809191/222151067-37987569-f05e-4ef4-b5bb-3cc221160399.png">

<img width="633" alt="image" src="https://user-images.githubusercontent.com/1809191/222150990-8c2761da-acc1-479b-8b75-3d95302afdca.png">

**Comment by tfranzel** [comment]
Haha, that's a good one. 🤣  Just wrong on multiple levels. And that, my dear friends, is why ChatGPT is not gonna replace us anytime soon. Don't get me wrong, it is an amazing feat of technology. However, people can anthropomorphize that thing all day long, it is still not gonna _think_.
